### PR TITLE
[TASK-1]: copy the test project for reviews

### PR DESCRIPTION
**Pull request by IvanMykytyn**

### CODE REVIEW

File: src/api/auth/api.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+import publicApi from "src/configs/publicApi";
+import type { IAuthResponse } from "./types";
+
+export const getAuthToken = async (): Promise<IAuthResponse> => {
+  return publicApi
+    .get("/auth/anonymous?platform=subscriptions")
+    .then((res) => res.data);
```
Comment (YuliiaDikun) [inline src/api/auth/api.ts:8]: you should use `async/await` syntax with `try catch` construction or `fetch` with `than catch` construction when you work with Promises.

File: src/api/auth/hooks.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+export const useAuthTokenMutation = () => {
+  const { setIsLogged } = useAuthStore();
+
+  return useMutation<IAuthResponse>({
+    mutationKey: ["auth"],
+    mutationFn: async () => {
+      return getAuthToken().then((data) => {
+        return data;
+      });
```
Comment (YuliiaDikun) [inline src/api/auth/hooks.tsx:16]: same. pay attention to error handling when you work with promise.

File: src/components/VideoPlayer/VideoPlayer.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+  src,
+  poster,
+  muted,
+  autoPlay,
+  width = "78%",
+  loop = false,
+  controls = true,
+  withPictureInPicture = false,
+}) => {
+  const videoRef = useRef<any>(null);
```
Comment (YuliiaDikun) [inline src/components/VideoPlayer/VideoPlayer.tsx]: maybe you could specify useRef type? [Read more](https://www.robinwieruch.de/typescript-react-useref/)

File: src/api/auth/api.ts (hunk 1 above)
Comment (IvanMykytyn) [reply src/api/auth/api.ts:8]: Fortunately, react-query does this for me

File: src/api/auth/hooks.tsx (hunk 2 above)
Comment (IvanMykytyn) [reply src/api/auth/hooks.tsx:16]: yeah, I know(( I haven't done error handling due to lack of time. I will try to find time to fix it. thx

File: src/components/Course/Course.tsx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+              <Center>
+                <Divider size='sm' orientation='vertical' sx={{ height: 15 }} />
+              </Center>
+
+              <Text color={"gray.7"} fz={"xs"}>
+                {getMinutes(duration)} minutes
+              </Text>
+
+              {containsLockedLessons && (
+                <>
```
Comment (perhamik) [inline src/components/Course/Course.tsx:86]: I remember that I have some troubles with this syntax:
**containsLockedLessons && <>...</>**
But, ternary operator (in my case) solved this:
**containsLockedLessons ? <>...</> : <></>**
Where "..." - children and <></> – just empty React.Fragment

File: src/components/Course/Course.tsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+              )}
+            </Group>
+            <Group pt={10}>
+              {tags.map((tag) => {
+                return (
+                  <Badge key={tag} color='gray' radius='sm'>
+                    {tag}
+                  </Badge>
+                );
+              })}
```
Comment (perhamik) [inline src/components/Course/Course.tsx]: We can minimize here, by removing figure brackets and "return"

File: src/api/courses/api.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import privateApi from "src/configs/private-api";
+import type { ICourses, ICourseWithLessons } from "./types";
+
+export const getCourses = async (): Promise<ICourses> => {
+  return privateApi.get("/core/preview-courses").then((res) => res.data);
+};
```
Comment (perhamik) [inline src/api/courses/api.ts:6]: Initially, we got **unhandled** error 401 and also Axios returns "ERR_BAD_REQUEST"

File: src/components/Lesson/Lesson.tsx [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+      sx={{
+        borderRadius: 5,
+        cursor: status === "locked" ? "not-allowed" : "pointer",
+      }}
+      onClick={() => {
+        status === "unlocked" && handleLessonChange(id);
+      }}
+    >
+      <Grid.Col span={1}>
+        <AspectRatio ratio={16 / 9} maw={400} mx='auto'>
```
Comment (perhamik) [inline src/components/Lesson/Lesson.tsx:52]: Aspect-ratio (as CSS rule) not working in Safari.
As the result, you'd have some layout issues.

File: src/pages/CourseItem/CourseItem.tsx [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+  const currentLesson = lessons?.find(
+    (lesson) => lesson.id === currentLessonId
+  );
+
+  useEffect(() => {
+    if (lessons && !currentLessonId) {
+      setCurrentLessonId(lessons?.[0].id || "");
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (perhamik) [inline src/pages/CourseItem/CourseItem.tsx:28]: Better to set it **.eslintrc.json** 
Cause, this rule isn't working well and the hints from exhaustive-deps are usually incorrect.

File: src/pages/Courses/Courses.tsx [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+      </Title>
+      <Group mt={15} spacing={"sm"}>
+        {courses?.map((course, index) => {
+          if (
+            index >= (activePage - 1) * COURSES_PER_PAGE &&
+            index < activePage * COURSES_PER_PAGE
+          ) {
+            return <Course key={course.id} {...course} />;
+          }
+          return null;
```
Comment (perhamik) [inline src/pages/Courses/Courses.tsx:47]: Better to split this array with courses.

**Review by perhamik — COMMENTED** [review]
Nice, but need to add Error Handlers

File: src/pages/Courses/Courses.tsx (hunk 9 above)
Comment (IvanMykytyn) [reply src/pages/Courses/Courses.tsx:47]: It's more like imitation of pagination than the real one. So I decided not to split array of courses here.

File: src/components/Course/Course.tsx (hunk 4 above)
Comment (IvanMykytyn) [reply src/components/Course/Course.tsx:86]: I don't understand the problem((I will try to investigate this later. But even so, I think containsLockedLessons ? <>...</> : null, will be more clear.

**Review by YuliiaDikun — APPROVED** [review]

**Review by perhamik — APPROVED** [review]

**Pull request merged**